libecl/src/ecl_nnc_data.c

```diff
@@ -0,0 +1,101 @@
+/*
+   Copyright (C) 2017  Statoil ASA, Norway.
+
+   The file 'ecl_file_view.c' is part of ERT - Ensemble based Reservoir Tool.
+
+   ERT is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or
+   FITNESS FOR A PARTICULAR PURPOSE.
+
+   See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+   for more details.
+*/
+
+#define ECL_NNC_DATA_TYPE_ID 83756236
+
+#include <ert/ecl/ecl_nnc_data.h>
+#include <ert/ecl/ecl_nnc_geometry.h>
+#include <ert/ecl/ecl_file.h>
+#include <ert/ecl/ecl_file_view.h>
+#include <ert/ecl/ecl_grid.h>
+#include <ert/ecl/ecl_kw_magic.h>
+
+
+struct ecl_nnc_data_struct {
+   UTIL_TYPE_ID_DECLARATION;
+   int size;
+   double * values;
+};
+
+
+
+static ecl_kw_type * ecl_nnc_data_get_trangl_kw( const ecl_file_view_type * init_file, int lgr_nr ) {
+  ecl_kw_type * tran_kw = NULL;
+  return tran_kw;
+}
+
+
+static void ecl_nnc_data_set_trans(ecl_nnc_data_type * data, ecl_nnc_geometry_type * nnc_geo, ecl_file_view_type * init_file) {
+ 
+   data->size = ecl_nnc_geometry_size( nnc_geo );
+   
+   data->values = util_malloc( data->size * sizeof(double));
```

Comment:
The `date->size = ... ` and `data->values = ...` assignments should be in the `ecl_nnc_data_alloc_tran( )` function.